features/filesystem/littlefs/littlefs/lfs.c

```diff
@@ -2040,8 +2040,8 @@ int lfs_mount(lfs_t *lfs, const struct lfs_config *cfg) {
     }
 
     // setup free lookahead
-    lfs->free.begin = -lfs->cfg->lookahead;
-    lfs->free.off = lfs->cfg->lookahead;
+    lfs->free.begin = -lfs_min(lfs->cfg->lookahead, lfs->cfg->block_count);
```

Comment:
Would be good to have a comment here as to why you are doing this ...

Reply:
Should be updated